src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,61 @@
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { currentTodoSlice } from '../../features/currentTodo';
+import { Todo } from '../../types/Todo';
+import cn from 'classnames';
+
+type Props = {
+  todo: Todo;
+};
+
+export const TodoItem: React.FC<Props> = ({ todo }) => {
+  const { id, title, completed } = todo;
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const dispatch = useAppDispatch();
+
+  const handleSetCurrent = (todoFromTable: Todo) => {
+    dispatch(currentTodoSlice.actions.setCurrentTodo(todoFromTable));
```

Comment:
Ensure that the `setCurrentTodo` action in `currentTodoSlice` is correctly defined to handle the `todoFromTable` object. Verify that the state update logic in the slice is correct.